core/src/network/event.rs

```diff
@@ -98,15 +98,18 @@ where
         /// Address used to send back data to the remote.
         send_back_addr: Multiaddr,
         /// The error that happened.
+        // TODO: This enum contains `PeerIdMismatch`, which cannot possibly happen for an incoming
+        // connection. Ideally we should use a different enum.
```

Comment:
That's not quite right because I collapsed the cases `PeerIdMismatch` and `LocalPeerId` into `PendingConnectionError::InvalidPeerId` (i.e. the remote identified with an invalid peer ID, which either does not match the expected one or is equal to the local peer ID and thus similarly unexpected, or otherwise invalid). If you consider the distinction(s) important, you may want to open an issue or add a todo about splitting these errors again.

Reply:
Oh right. Sorry, like yesterday I still have the automatism of the old code.